### ISSUE TITLE
Crash after decrypt of old apikey (from another machine)

### ISSUE THREAD

**Issue by kingpin2k** [NuGetPackageExplorer/NuGetPackageExplorer#73]
I'm in Windows 10 x64.  (I manually uploaded it, if the nupkg will help https://www.nuget.org/packages/HTMLBarsHelper/2.0.0.1)

```
Application: NuGetPackageExplorer.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: System.Security.Cryptography.CryptographicException
   at System.Security.Cryptography.ProtectedData.Unprotect(Byte[], Byte[], System.Security.Cryptography.DataProtectionScope)
   at NuGet.SettingsExtensions.GetDecryptedValue(NuGet.ISettings, System.String, System.String)
   at PackageExplorer.SettingsManager.ReadApiKey(System.String)
   at PackageExplorerViewModel.PublishPackageViewModel.set_SelectedPublishItem(System.String)
   at PackageExplorerViewModel.PublishPackageViewModel..ctor(PackageExplorerViewModel.MruPackageSourceManager, NuGetPackageExplorer.Types.ISettingsManager, PackageExplorerViewModel.PackageViewModel)
   at PackageExplorerViewModel.PackageViewModel.PublishExecute()
   at PackageExplorerViewModel.RelayCommand.Execute(System.Object)
   at MS.Internal.Commands.CommandHelpers.CriticalExecuteCommandSource(System.Windows.Input.ICommandSource, Boolean)
   at System.Windows.Controls.MenuItem.InvokeClickAfterRender(System.Object)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(System.Delegate, System.Object, Int32)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(System.Object, System.Delegate, System.Object, Int32, System.Delegate)
   at System.Windows.Threading.DispatcherOperation.InvokeImpl()
   at System.Windows.Threading.DispatcherOperation.InvokeInSecurityContext(System.Object)
   at System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object)
   at System.Windows.Threading.DispatcherOperation.Invoke()
   at System.Windows.Threading.Dispatcher.ProcessQueue()
   at System.Windows.Threading.Dispatcher.WndProcHook(IntPtr, Int32, IntPtr, IntPtr, Boolean ByRef)
   at MS.Win32.HwndWrapper.WndProc(IntPtr, Int32, IntPtr, IntPtr, Boolean ByRef)
   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(System.Object)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(System.Delegate, System.Object, Int32)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(System.Object, System.Delegate, System.Object, Int32, System.Delegate)
   at System.Windows.Threading.Dispatcher.LegacyInvokeImpl(System.Windows.Threading.DispatcherPriority, System.TimeSpan, System.Delegate, System.Object, Int32)
   at MS.Win32.HwndSubclass.SubclassWndProc(IntPtr, Int32, IntPtr, IntPtr)
   at MS.Win32.UnsafeNativeMethods.DispatchMessage(System.Windows.Interop.MSG ByRef)
   at System.Windows.Threading.Dispatcher.PushFrameImpl(System.Windows.Threading.DispatcherFrame)
   at System.Windows.Threading.Dispatcher.PushFrame(System.Windows.Threading.DispatcherFrame)
   at System.Windows.Application.RunDispatcher(System.Object)
   at System.Windows.Application.RunInternal(System.Windows.Window)
   at System.Windows.Application.Run(System.Windows.Window)
   at PackageExplorer.App.Main()

```

**Comment by 304NotModified** [comment]
never seen this exception, this is a publish to nuget.org? Is it as nuget v2 or v3 package?

**Comment by kingpin2k** [comment]
Correct, on publish to nuget.org, I'm using 3.13 now and still seeing the same issue. And yes, it's a v2 package.

**Comment by 304NotModified** [comment]
Is the api key correct? Never seen this exception before in npe

**Comment by kingpin2k** [comment]
It crashes immediately when I press the publish.  I just pulled down the source and ran it, and like the stack trace says, it crashes trying to decrypt the saved api key.  I wouldn't have every copied the nuget.config from one machine to another, but it's entirely possible when I installed windows 10 (fresh install) I might have started out as a local account and then allowed it to switch me to a microsoft account.  And I'd guess during that conversion it may have copied the appdata for nuget. Since it's using DataProtectionScope.CurrentUser, which makes sense, and since it was possibly first saved under the local account, it isn't matching, so it's failing to read it using the microsoft account.

This is just purely speculation, but it makes sense to me.

Regardless, after ripping the api key out of the config it was able to continue.  My guess is this isn't a huge scenario, but it could potentially affect people who upgrade, but I'm just speculating.  At the least, it could just be an attempt to decrypt, and on failure return nothing, or remove the api key.  

    <add key="https://www.nuget.org" value="Value Removed" />
